src/Build/Definition/ProjectCollection.cs

```diff
@@ -650,26 +661,14 @@ public bool IsBuildEnabled
             [DebuggerStepThrough]
             get
             {
-                using (_locker.EnterDisposableUpgradeableReadLock())
-                {
-                    return _isBuildEnabled;
-                }
+                return _isBuildEnabled != 0;
             }
 
             [DebuggerStepThrough]
             set
             {
-                bool sendEvent = false;
-                using (_locker.EnterDisposableWriteLock())
-                {
-                    if (_isBuildEnabled != value)
-                    {
-                        _isBuildEnabled = value;
-                        sendEvent = true;
-                    }
-                }
-
-                if (sendEvent)
+                int intValue = value ? 1 : 0;
+                if (Interlocked.Exchange(ref _isBuildEnabled, intValue) != intValue)
```

Comment:
Playing it safe with the flags to guarantee that `OnProjectCollectionChanged` is delivered only once if multiple threads get here.